### PR TITLE
Update Codelabs for 3.16

### PR DESCRIPTION
**Pull request by domesticmouse**
## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by domesticmouse** [comment]
Looks like Windows is failing to load DLLs. Argh.

```console
Invalid argument(s): Failed to load dynamic library
  'd:\a\codelabs\codelabs\ffigen_codelab\step_07\example\build\windows\runner\debug\ffigen_app.dll':
  error code 126
```

File: ffigen_codelab/step_07/lib/ffigen_app.dart [hunk 1]
```diff
@@ -30,7 +30,7 @@ final DynamicLibrary _dylib = () {
   if (Platform.isWindows) {
     if (Platform.environment.containsKey('FLUTTER_TEST')) {
       return DynamicLibrary.open(p.canonicalize(
-          p.join(r'build\windows\runner\Debug', '$_libName.dll')));
+          p.join(r'build\windows\x64\runner\Debug', '$_libName.dll')));
```
Comment (domesticmouse) [inline ffigen_codelab/step_07/lib/ffigen_app.dart:33]: @loic-sharma Looks like we are now generating CPU architecture as part of our library build paths. Is there a way I can get that architecture at run time, so I can make this test architecture agnostic?

Comment (loic-sharma) [reply ffigen_codelab/step_07/lib/ffigen_app.dart:33]: Yup, `Abi.current`: https://api.dart.dev/stable/3.0.3/dart-ffi/Abi/Abi.current.html

For more info, see: https://flutter.dev/go/windows-arm64, there's a section on bundling pre-compiled code.

Comment (loic-sharma) [reply ffigen_codelab/step_07/lib/ffigen_app.dart:33]: `Abi.current` can be used in the app's logic to determine the current architecture. However, be careful in compilation/bundling scenarios: the current architecture can be different than the target architecture. If you're using CMake to bundle a pre-compiled library, you should use `FLUTTER_TARGET_PLATFORM` instead.

Comment (domesticmouse) [reply ffigen_codelab/step_07/lib/ffigen_app.dart]: I haven't touched bundling in this codelab. At this point I'm just testing that a DLL can be linked and then accessed via FFI.

**Review by RedBrogdon — APPROVED** [review]

**Pull request merged**